GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/AnimationInteractivityManager.java

```diff
@@ -628,17 +663,122 @@ public void onSensorEvent(SensorEvent event) {
+                                            GVRSceneObject inlineSceneObject = inlineObject.getInlineGVRSceneObject();
+                                            GVRAnimator animator = (GVRAnimator) inlineSceneObject.getComponent(GVRAnimator.getComponentType());
+                                            if ( animator != null ) {
+                                                List<GVRAnimation> gvrAnimations = animator.getAnimationList();
+                                                GVRAnimation gvrAnimation = null;
+
+                                                if  ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) || (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                    if ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) && (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                        // use the original FBX animation.
+                                                        gvrAnimation = gvrAnimations.get(0);
+                                                    }
+                                                    else {
+                                                        Log.e(TAG, "Error: ROUTE specifying the keyFrame range - only 1 may be an integers");
+                                                    }
+                                                }
+                                                else {
+                                                    // FBX animation specified in ROUTE as a set of keyFrames
+                                                    // play only a subset of the animation based on keyframes
+                                                    gvrAnimation = interactiveObjectFinal.getFBXAnimation();
+                                                    // check if we already copied over the animation
+                                                    if (gvrAnimation == null) {
+                                                        GVRAnimation gvrOriginalAnimation = gvrAnimations.get(0);
+                                                        if (gvrOriginalAnimation instanceof GVRKeyFrameAnimation) {
+                                                            GVRKeyFrameAnimation gvrOriginalKeyFrameAnimation = (GVRKeyFrameAnimation) gvrOriginalAnimation;
+                                                            GVRAnimationChannel gvrOriginalAnimationChannel = gvrOriginalKeyFrameAnimation.getChannel(0);
+
+                                                            try {
+                                                                // copy the original animation, then save just the requested frames
+                                                                // and their translation, scale and rotate
```

Comment:
What if you could modify the existing animation instead of creating a new one? Could you please explain what is going in the block and why?